lib/acts_as_taggable_on/taggable/core.rb

```diff
@@ -28,26 +29,43 @@ def initialize_acts_as_taggable_on_core
             has_many context_taggings, -> { includes(:tag).order(taggings_order).where(context: tags_type) },
                      as: :taggable,
                      class_name: 'ActsAsTaggableOn::Tagging',
-                     dependent: :destroy
+                     dependent: :destroy,
+                     after_add: :dirtify_tag_list,
+                     after_remove: :dirtify_tag_list
 
             has_many context_tags, -> { order(taggings_order) },
                      class_name: 'ActsAsTaggableOn::Tag',
                      through: context_taggings,
                      source: :tag
+
+            attribute "#{tags_type.singularize}_list".to_sym, ActiveModel::Type::Value.new
           end
 
           taggable_mixin.class_eval <<-RUBY, __FILE__, __LINE__ + 1
+
             def #{tag_type}_list
               tag_list_on('#{tags_type}')
             end
 
             def #{tag_type}_list=(new_tags)
+              parsed_new_list = ActsAsTaggableOn.default_parser.new(new_tags).parse
+
+              if self.class.preserve_tag_order? || parsed_new_list.sort != #{tag_type}_list.sort
+                set_attribute_was('#{tag_type}_list', #{tag_type}_list)
+                write_attribute("#{tag_type}_list", parsed_new_list)
+              end
+
               set_tag_list_on('#{tags_type}', new_tags)
             end
 
             def all_#{tags_type}_list
               all_tags_list_on('#{tags_type}')
             end
+
+            private
+            def dirtify_tag_list tagging
```

Comment:
You should add parenthesis for the parameters, it seems to be what's done in this project. 
```ruby
def dirtify_tag_list(tagging)
  ...
end
```